### DISCUSSION TITLE
Can I add custom data without using input.

### DISCUSSION THREAD

**Discussion by marcmok6696** [react-hook-form/react-hook-form#2082] [Q&A] [answered]
I am currently struggling in adding extra data to the form.For example, i want the input type also stored into the data.
my code is this
**<Controller  as={<TextField />} fullWidth color = 'primary'  margin="none" required id="standard-basic" name={`questionList[${index}].question`} control={control} />**

I want to add a field under questionList[0], like  questionList[0].type = textfield.
Is there a way to do it?

**Comment by bluebill1049** [comment, marked as answer]
you mean custom register at useEffect and using setValue to update the input value.

**Reply by marcmok6696** [reply]
Thanks for your reply. But I think no.
What I want to do is I have a form and different buttons to let user chooses. **Those buttons are not within  the form tag**. for example a textfield button will generate a session with a textfield input component, like google form or Microsoft form.  And instead of only storing the input value into the form data, i want to also store the question type that the user choose (e.g. {question[0].type:"textfield"}
I am not sure the way to do it.

**Reply by bluebill1049** [reply]
as long as you have register those buttons, they should worked.

**Reply by marcmok6696** [reply]
I have read through the document but I am not sure how to register the button. I am using the controller tag as I want to use material UI.

**Reply by bluebill1049** [reply]
```
regsiter('test');

setValue('test', 'data')
```

**Reply by marcmok6696** [reply]
Would you mind to provide some examples? I am sorry but i still can't figure out how to do it

**Reply by bluebill1049** [reply]
```
useEffect(() => {
  regsiter('test');
  setValue('test', 'data')
}. [register, setValue])
```

those values should be part of the submit data now.
